app/javascript/components/retirement-form/index.jsx

```diff
@@ -0,0 +1,68 @@
+import React, { useEffect, useState } from 'react';
+import moment from 'moment';
+
+import MiqFormRenderer from '@@ddf';
+import createSchema from './retirement-form.schema';
+import handleFailure from '../../helpers/handle-failure';
+
+const RetirementForm = ({ retirementID }) => {
+  const id = retirementID.replace(/\D/g,'');
+  const [{ initialValues, isLoading }, setState] = useState({
+    isLoading: !!id,
+    initialValues: { formMode: '' },
+  });
+
+  const cancelUrl = `/${ManageIQ.controller}/retire?button=cancel`;
+  const saveURL = `/${ManageIQ.controller}/retire?button=save`;
+
+  const onSubmit = (data) => {
```

Comment:
Destructuring is your friend, you can definitely pull out the `formMode` here, but I guess all the other `date.` calls can be moved up here...